subsys/bluetooth/audio/ascs.c

```diff
@@ -1425,22 +1424,27 @@ static int ase_stream_qos(struct bt_audio_stream *stream,
 		}
 	}
 
-	iso = audio_iso_get_or_new(ascs, cig_id, cis_id);
-	if (iso == NULL) {
-		BT_ERR("Could not allocate audio_iso");
-		return -ENOMEM;
-	}
+	/* In QoS->QoS transition, the EP already owns audio ISO object */
+	if (ep->iso == NULL) {
+		struct bt_audio_iso *iso;
+
+		iso = audio_iso_get_or_new(ascs, cig_id, cis_id);
+		if (iso == NULL) {
+			BT_ERR("Could not allocate audio_iso");
+			return -ENOMEM;
+		}
 
-	if (bt_audio_iso_get_ep(iso, ep->dir) != NULL) {
-		BT_ERR("iso %p already in use in dir %u",
-		       &iso->chan, ep->dir);
+		if (bt_audio_iso_get_ep(iso, ep->dir) != NULL) {
+			BT_ERR("iso %p already in use in dir %u",
+			&iso->chan, ep->dir);
+			bt_audio_iso_unref(iso);
+			return -EALREADY;
+		}
+
+		bt_audio_iso_bind_ep(iso, ep);
```

Comment:
If going `QoS->QoS`, isn't there a chance the CIS and CIG ID changed?

Should we instead unbind if `ep->iso != NULL`, and then get a new audio ISO afterwards? 

Or at least verify that the CIS and CIG ID didn't change?

Reply:
Eh, that is true, I did it initially like you suggested. I don't remember why I decided to remove the check TBH.

Reply:
Oh, I remember, It's because the ISO channel is not connected (unused) yet at this point